src/System.IO.FileSystem/src/System/IO/DirectoryInfo.cs

```diff
@@ -180,8 +180,7 @@ private FileInfo[] InternalGetFiles(String searchPattern, SearchOption searchOpt
             Contract.Requires(searchOption == SearchOption.AllDirectories || searchOption == SearchOption.TopDirectoryOnly);
 
             IEnumerable<FileInfo> enble = (IEnumerable<FileInfo>)FileSystem.Current.EnumerateFileSystemInfos(FullPath, searchPattern, searchOption, SearchTarget.Files);
-            List<FileInfo> fileList = new List<FileInfo>(enble);
-            return fileList.ToArray();
+            return EnumerableHelpers.ToArray(enble);
```

Comment:
Completely unrelated, but 'enble' is a silly variable name. Took me a minute to parse as I kept reading it as enable misspelled. :)